src/Microsoft.ML.FastTree/TreeEnsemble/TreeEnsembleCombiner.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.FastTree.Internal;
+using Microsoft.ML.Runtime.Internal.Calibration;
+
+[assembly: LoadableClass(typeof(TreeEnsembleCombiner), null, typeof(SignatureModelCombiner), "Fast Tree Model Combiner", "FastTreeCombiner")]
+
+namespace Microsoft.ML.Runtime.FastTree.Internal
+{
+    public sealed class TreeEnsembleCombiner : IModelCombiner<IPredictorProducing<float>, IPredictorProducing<float>>
+    {
+        private readonly IHost _host;
+        private readonly PredictionKind _kind;
+
+        public TreeEnsembleCombiner(IHostEnvironment env, PredictionKind kind)
+        {
+            _host = env.Register("TreeEnsembleCombiner");
+            switch (kind)
+            {
+                case PredictionKind.BinaryClassification:
+                case PredictionKind.Regression:
+                case PredictionKind.Ranking:
+                    _kind = kind;
+                    break;
+                default:
+                    throw _host.ExceptUserArg(nameof(kind), "Tree ensembles can be either binary classifiers, regressors or rankers");
```

Comment:
>binary classifiers, regressors or rankers [](start = 90, length = 41)

Would it be better for this to be an interpolated `$` style string with `{nameof(PredictionKind.BinaryClassificiation)}`, and so on? Not sure.